### PR TITLE
#77 Tests for OnMoon\OpenApiServerBundle\CodeGenerator\Definitions

### PR DESCRIPTION
**Pull request by Alex2k**

### CODE REVIEW

**Comment by sspat** [comment]
@Alex2k tests failing after merge with master

File: test/unit/CodeGenerator/Definitions/PropertyDefinitionTest.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        Assert::assertSame($payload['objectTypeDefinition'], $propertyDefinition->getObjectTypeDefinition());
+        Assert::assertSame($payload['nullable'], $propertyDefinition->isNullable());
+        Assert::assertSame($payload['getterName'], $propertyDefinition->getGetterName());
+        Assert::assertSame($payload['setterName'], $propertyDefinition->getSetterName());
+        Assert::assertSame($payload['hasGetter'], $propertyDefinition->hasGetter());
+        Assert::assertSame($payload['hasSetter'], $propertyDefinition->hasSetter());
+        Assert::assertSame($payload['inConstructor'], $propertyDefinition->isInConstructor());
+        Assert::assertSame($expected['specProperty']['name'], $propertyDefinition->getSpecPropertyName());
+        Assert::assertSame($expected['specProperty']['isArray'], $propertyDefinition->isArray());
+        Assert::assertSame($expected['specProperty']['scalarTypeId'], $propertyDefinition->getScalarTypeId());
```
Comment (sspat) [inline test/unit/CodeGenerator/Definitions/PropertyDefinitionTest.php:104]: getDescription method is not tested

File: test/unit/CodeGenerator/Definitions/RequestBodyDtoDefinitionTest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\PropertyDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\RequestBodyDtoDefinition;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Property;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\RequestBodyDtoDefinition
+ */
+final class RequestBodyDtoDefinitionTest extends TestCase
```
Comment (sspat) [inline test/unit/CodeGenerator/Definitions/RequestBodyDtoDefinitionTest.php:16]: This test is not needed, RequestBodyDtoDefinition is an empty class and DtoDefinition has its own test.

File: test/unit/CodeGenerator/Definitions/RequestParametersDtoDefinitionTest.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\PropertyDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\RequestParametersDtoDefinition;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Property;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\RequestParametersDtoDefinition
+ */
+final class RequestParametersDtoDefinitionTest extends TestCase
```
Comment (sspat) [inline test/unit/CodeGenerator/Definitions/RequestParametersDtoDefinitionTest.php:16]: This test is not needed, RequestParametersDtoDefinition is an empty class and DtoDefinition has its own test.

File: test/unit/CodeGenerator/Definitions/ResponseDtoDefinitionTest.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        ];
+    }
+
+    /**
+     * @param mixed[] $conditions
+     * @param mixed[] $expected
+     *
+     * @dataProvider responseDtoDefinitionProvider
+     */
+    public function testResponseDtoDefinition(array $conditions, array $expected): void
```
Comment (sspat) [inline test/unit/CodeGenerator/Definitions/ResponseDtoDefinitionTest.php:41]: Only getStatusCode needs to be tested, all other methods of the class are inherited and final.

Comment (Alex2k) [reply test/unit/CodeGenerator/Definitions/ResponseDtoDefinitionTest.php:41]: Also we need to check the dispatch of parent properties in the constructor.

**Pull request merged**